### DISCUSSION TITLE
HiDPi 144dpi 1.5 scaling

### DISCUSSION THREAD

**Discussion by bat17v** [hyprwm/Hyprland#7923] [Q&A] [answered]
I have a monitor with 144 dpi. So, I need to scale all element to 1.5. It perfectly works by wayland. But on Xorg apps looks pixeled. I know that I can turn off scaling xwayland by adding to config:
```
xwayland {
    force_zero_scaling = true
}
```
Add then I need to scale element using toolkit-specific methods, but I cannot found how can I configure fractional scaling in GDK?

**Comment by fufexan** [comment]
GDK does not support fractional scaling.

**Reply by bat17v** [reply]
But KDE Plasma scales GDK apps. How does it work?

**Reply by fufexan** [reply]
I don't know. Probably by scaling xwayland entirely, which is not possible in Hyprland (without blurriness).

**Comment by tannerellen** [comment, marked as answer]
The best way I have found to scale xwayland apps is to adjust the dpi settings for them. 
Make sure hyprland has:

xwayland {
    force_zero_scaling = true
}

You need to install xorg-xrdb and then specify a dpi setting in ~/.Xresources
My current config is:

```
Xft.dpi: 180
Xft.autohint: 0
Xft.lcdfilter:  lcddefault
Xft.hintstyle:  hintfull
Xft.hinting: 1
Xft.antialias: 1
Xft.rgba: rgb
```

The base dpi is 96. So you would multiply that by whatever your scale is to get your dpi. So if your scale is 1.5:
96 * 1.5 = 144

`Xft.dpi: 144`

It's not perfect but it's way better than blurry.

**Reply by noahfraiture** [reply]
Hey, i followed you steps  but my xwayland app aren't scaled, no matter de dpi value. Is there something else to do ?

**Reply by tannerellen** [reply]
If you installed xorg-xrdb it should work. You can verify the current settings by running the command: `xrdb -query -all`

It should list your settings like Xft.dpi like the ones I listed.  If it doesn't list your settings you may need to load the xresources file:

`xrdb ~/.Xresources`

I use SDDM as my display manager and I'm pretty sure it does that for you but if you don't use a display manager or it isn't working try the above command and see.

Here is the arch wiki on this which has some usefull info:

https://wiki.archlinux.org/title/X_resources

**Reply by noahfraiture** [reply]
Oh yeah thx, I forgot to load it and restart the application, it should be closed as answered !

**Comment by Megadash452** [comment]
Hey @tannerellen, thanks for the insight. All my XWayland apps are scaled properly with this trick :)

If I put the command in my `.bashrc` and `.profile`, it returns with an error: `xrdb: Can't open display ''`. Instead, it should be ran in `hyprland.conf` like this: `exec-once xrdb ~/.Xresources`.

Just wanted to put this information out there in hopes that it helps someone. It definitely would've saved me hours of debuging :pensive:
